File: apps/android/core/designsystem/src/main/kotlin/app/gauja/core/designsystem/GaujaTheme.kt
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
package app.gauja.core.designsystem

import androidx.compose.material3.MaterialTheme
import androidx.compose.runtime.Composable
import app.gauja.core.designsystem.generated.GaujaDarkColorScheme
import app.gauja.core.designsystem.generated.GaujaLightColorScheme
import app.gauja.core.designsystem.generated.GaujaShapes
import app.gauja.core.designsystem.generated.GaujaTypography

@Composable
fun GaujaTheme(darkTheme: Boolean = true, content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = if (darkTheme) GaujaDarkColorScheme else GaujaLightColorScheme,
        typography = GaujaTypography,
        shapes = GaujaShapes,
        content = content,
    )
}

File: apps/ios/Packages/DesignSystem/Sources/DesignSystem/GaujaTheme.swift
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
import SwiftUI

public struct GaujaTheme<Content: View>: View {
    private let content: Content
    private let scheme: ColorScheme

    public init(scheme: ColorScheme = .dark, @ViewBuilder content: () -> Content) {
        self.scheme = scheme
        self.content = content()
    }

    public var body: some View {
        content.tint(.gaujaPrimary(scheme))
            .background(Color.gaujaBackground(scheme))
            .preferredColorScheme(scheme)
    }
}
